### PR TITLE
Troca de senha do usuario autenticado (issue #93)

### LINKED ISSUE #93: Troca de senha do usuário (frontend)
## Descrição

Hoje o usuário autenticado não possui, no frontend, um fluxo para alterar a própria senha. Como o backend já entregou o endpoint correspondente (ver [carlessopilatesapi#78](https://github.com/FabioCarlesso/carlessopilatesapi/issues/78)), falta consumir essa API no Angular e oferecer uma experiência segura e clara para o usuário trocar sua senha quando quiser ou quando suspeitar de comprometimento.

## Objetivo

Disponibilizar no frontend um fluxo de troca de senha para o usuário autenticado, integrado ao endpoint do backend, com validações em tela coerentes com a política de segurança da aplicação e feedback adequado de sucesso/erro.

## Escopo da implementação

- Nova tela / seção **"Alterar senha"** acessível a partir do menu do usuário autenticado (ex.: perfil / configurações da conta).
- Formulário com os campos:
  - Senha atual
  - Nova senha
  - Confirmação da nova senha
- Validações em tempo real no formulário (Reactive Forms):
  - Todos os campos obrigatórios.
  - Política mínima da nova senha (tamanho/complexidade) coerente com a regra aplicada pelo backend.
  - Nova senha e confirmação devem coincidir.
  - Nova senha deve ser diferente da senha atual.
- Botão de submit desabilitado enquanto o formulário estiver inválido ou em loading.
- Toggle de visibilidade (mostrar/ocultar) para cada campo de senha.
- Integração com o endpoint do backend (ex.: `PUT /usuarios/me/senha`) através de um service Angular dedicado.
- Tratamento de respostas:
  - **Sucesso:** mensagem de confirmação, limpar formulário e, conforme estratégia de autenticação, redirecionar para login se a sessão/token for invalidada.
  - **Erro de senha atual incorreta:** mensagem específica no campo "Senha atual".
  - **Erros de validação do backend:** exibir mensagens vindas da API próximas ao campo correspondente.
  - **Erros genéricos / falha de rede:** snackbar/toast informando falha sem expor detalhes sensíveis.
- Aderência ao **Design System** já existente (tokens, dark mode, componentes do projeto).
- Acessibilidade básica: labels associados, navegação por teclado, mensagens de erro lidas por leitores de tela.

## Critérios de aceite

- [ ] Usuário autenticado consegue acessar a tela de troca de senha a partir da navegação do app.
- [ ] Formulário valida em tela: obrigatoriedade, política da nova senha, coincidência da confirmação e diferença em relação à senha atual.
- [ ] Submit chama o endpoint do backend apenas com o formulário válido.
- [ ] Em caso de sucesso, é exibida mensagem de confirmação e o usuário é tratado conforme a estratégia de autenticação (relogin ou manutenção da sessão).
- [ ] Em caso de senha atual incorreta, o erro é exibido de forma clara e localizada no campo correto.
- [ ] Erros vindos da API são apresentados sem expor informação sensível.
- [ ] Senhas nunca aparecem em logs do console nem em URLs (somente no corpo da requisição HTTPS).
- [ ] A tela funciona corretamente em **light** e **dark mode** e é responsiva.

## Sugestão de validação/testes

- Testes unitários do componente/formulário:
  - Validators de obrigatoriedade, política mínima, match entre nova senha e confirmação, nova ≠ atual.
  - Estado do botão de submit conforme validade e loading.
- Testes unitários do service:
  - Chamada para o endpoint correto, com método e payload esperados.
  - Tratamento de respostas de sucesso e erros conhecidos (401/400/500).
- Teste manual / e2e do fluxo completo:
  - Troca bem-sucedida e login subsequente com a nova senha.
  - Senha atual incorreta → mensagem adequada.
  - Nova senha fora da política → bloqueio no submit.
  - Confirmação divergente → bloqueio no submit.
  - Tentativa de reutilizar a senha atual como nova → bloqueio.
  - Verificar comportamento em dark mode e em telas pequenas.

## Observações técnicas

- Consumir o endpoint definido no backend em [carlessopilatesapi#78](https://github.com/FabioCarlesso/carlessopilatesapi/issues/78); confirmar contrato (rota, payload, códigos de erro) antes da implementação.
- Reaproveitar componentes de input/senha, botões, feedbacks (snackbar/toast) e tokens do Design System existente.
- Centralizar a chamada num service dedicado (ex.: `UsuarioService.alterarSenha(...)`), evitando lógica HTTP no componente.
- Nunca logar valores de senha (atual ou nova) no console em nenhum ambiente.
- Avaliar, junto ao backend, se a troca invalida o token atual: se sim, limpar o estado de autenticação e redirecionar para a tela de login após o sucesso.
- Mensagens de erro genéricas o suficiente para não vazar informação sensível (ex.: não diferenciar "usuário não existe" de "senha incorreta").

## Labels sugeridas

`frontend`, `enhancement`, `security`, `angular`, `ux`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo

- Nova tela `/perfil/alterar-senha` (`AlterarSenhaComponent`), protegida por `authGuard`, com formulario reativo `senhaAtual` / `novaSenha` / `confirmacaoNovaSenha`.
- Validacoes locais: obrigatoriedade, minimo de 8 caracteres na nova senha, confirmacao coincidente, nova senha diferente da atual. Toggle de visibilidade por campo (`aria-pressed` / `aria-label`).
- `UsuarioAdminService.alterarSenha` envia `PUT /api/users/me/senha` com `{ senhaAtual, novaSenha, confirmacaoNovaSenha }`; respostas 401/403 mapeiam para erro localizado no campo `senhaAtual`, 400 propaga `error.message` quando disponivel, demais erros caem em mensagem generica.
- Apos sucesso a tela exibe confirmacao, limpa o formulario, chama `AuthService.clearSession()` e redireciona para `/login` para relogin.
- Link **Alterar senha** adicionado a navbar do `AppComponent` para todo usuario autenticado.

## Motivo

Implementa o requisito de seguranca da issue #93: usuario autenticado precisa poder trocar a propria senha quando quiser ou quando suspeitar de comprometimento, consumindo o endpoint disponibilizado pelo backend (FabioCarlesso/carlessopilatesapi#78).

## Testes executados

- `npx ng test --watch=false --browsers=ChromeHeadless` → 574 SUCCESS
- `npm run build` → bundle gerado com sucesso (chunk `alterar-senha-component` em lazy load)

## Arquivos principais

- `src/app/pages/perfil/alterar-senha/alterar-senha.component.{ts,html,scss,spec.ts}` (novo)
- `src/app/core/services/usuario-admin.service.ts` / `.spec.ts` — novo metodo `alterarSenha`
- `src/app/core/models/auth.ts` — `AlterarSenhaRequestDTO`
- `src/app/app.routes.ts` / `.spec.ts` — rota `/perfil/alterar-senha`
- `src/app/app.component.html` / `.spec.ts` — link na navbar
- `src/styles.scss` — estilo `.btn-alterar-senha` herdando o padrao do botao da navbar
- `README.md`, `docs/context.md` — documentacao atualizada

## Referencia

Closes #93

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/app/pages/perfil/alterar-senha/alterar-senha.component.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        },
+        error: (err: HttpErrorResponse) => {
+          this.aplicarErro(err);
+          this.salvando = false;
+        }
+      });
+  }
+
+  private aplicarErro(err: HttpErrorResponse): void {
+    if (err?.status === 401 || err?.status === 403) {
```
Comment (FabioCarlesso) [inline src/app/pages/perfil/alterar-senha/alterar-senha.component.ts]: Problem: `401` and `403` are always mapped to `Senha atual incorreta.` and to a field-level `senhaAtual` error.

Why it matters: those statuses can also represent an expired/invalid token, an inactive user, a missing authorization rule, or another auth failure from Spring Security. In those cases the UI shows the wrong cause and may fight the global auth flow handled by `authInterceptor`, which is both a UX issue and an API-integration/security ambiguity.

Suggested improvement: only mark `senhaAtual` as incorrect when the backend returns a specific password-validation signal, such as a dedicated error code/body field or a 400/422 field error for `senhaAtual`. Let token/session failures be handled by the interceptor or show a generic authorization/session message.

Severity: Medium

File: src/app/app.component.html [hunk 2]
```diff
@@ -13,6 +13,7 @@
       }
     </ul>
     <div class="navbar-actions">
+      <a routerLink="/perfil/alterar-senha" class="btn-alterar-senha">Alterar senha</a>
```
Comment (FabioCarlesso) [inline src/app/app.component.html:16]: Problem: adding this third navbar action can overflow on narrow mobile widths because `.navbar-actions` stays a single non-wrapping row, while all three action buttons use `white-space: nowrap`.

Why it matters: at small widths, `Alterar senha` + `Tema claro/escuro` + `Sair` can exceed the available navbar width after padding, causing clipped or horizontally overflowing controls.

Suggested improvement: update the responsive navbar styles so `.navbar-actions` wraps or stacks under the existing `@media (max-width: 768px)` rule, and add a small component/style test or manual viewport check for the authenticated navbar.

Severity: Medium

**Review by FabioCarlesso — COMMENTED** [review]
Reviewed the Angular PR across documentation, bugs/edge cases, security, performance, Angular practices, API integration, UI/UX, tests, and code quality.

Documentation was updated in README and docs/context.md and generally matches the new feature. Tests/build pass locally: `npm test -- --watch=false --browsers=ChromeHeadless` (574 SUCCESS) and `npm run build`.

Areas that look good: route protection with `authGuard`, lazy loading, typed service method for the new endpoint, no unsafe HTML rendering, no exposed secrets/logging of passwords, double-submit guard, and focused service/component coverage. I left comments for the concrete issues I found.

**Comment by FabioCarlesso** [comment]
Correções do review aplicadas no commit `51ad78f`:

- O `AlterarSenhaComponent` agora só marca `senhaAtual` como incorreta quando a resposta do backend traz sinal explícito por `code`, `field` ou `fieldErrors` relacionado a `senhaAtual`.
- Respostas `401`/`403` sem esse sinal passam a exibir mensagem de autorização/sessão, evitando confundir token/permissão com senha atual incorreta.
- A navbar agora permite quebra das ações em telas estreitas para evitar overflow com `Alterar senha`, tema e sair.
- README e `docs/context.md` atualizados para refletir o novo comportamento.

Validação executada:

- `npm test -- --watch=false --browsers=ChromeHeadless` → 575 SUCCESS
- `npm run build` → sucesso

Não há script de lint configurado no `package.json`.
